File: 23.py
# Falling Tombstones - Python solution
from collections import deque
import sys

def simulate_falling(s, T):
    n = len(s)
    standing = [c == '1' for c in s]
    cnt = [0]*n
    for i in range(n):
        if i-1 >= 0 and standing[i-1]:
            cnt[i] += 1
        if i+1 < n and standing[i+1]:
            cnt[i] += 1
    q = deque()
    for i in range(n):
        if standing[i] and cnt[i] < T:
            q.append(i)
    days = 0
    while q:
        days += 1
        next_q = deque()
        # process all current failures simultaneously
        while q:
            i = q.popleft()
            if not standing[i]:
                continue
            standing[i] = False
            # update neighbors
            for nb in (i-1, i+1):
                if 0 <= nb < n and standing[nb]:
                    cnt[nb] -= 1
                    if cnt[nb] < T:
                        next_q.append(nb)
        q = next_q
    final = ''.join('1' if x else '0' for x in standing)
    return final, days

if __name__ == "__main__":
    data = sys.stdin.read().strip().split()
    if not data:
        s = "1101101"; T = 1
        final, days = simulate_falling(s, T)
        print(final)
        print(days)
    else:
        s = data[0].strip()
        T = int(data[1])
        final, days = simulate_falling(s, T)
        print(final)
        print(days)

File: 23.js
// Falling Tombstones - JavaScript
const fs = require('fs');

function simulateFalling(s, T) {
  const n = s.length;
  const standing = Array.from(s).map(c => c === '1');
  const cnt = new Array(n).fill(0);
  for (let i = 0; i < n; ++i) {
    if (i-1 >= 0 && standing[i-1]) cnt[i]++;
    if (i+1 < n && standing[i+1]) cnt[i]++;
  }
  let q = [];
  for (let i = 0; i < n; ++i) {
    if (standing[i] && cnt[i] < T) q.push(i);
  }
  let days = 0;
  while (q.length > 0) {
    days++;
    const next = [];
    for (const i of q) {
      if (!standing[i]) continue;
      standing[i] = false;
      for (const nb of [i-1, i+1]) {
        if (nb >= 0 && nb < n && standing[nb]) {
          cnt[nb]--;
          if (cnt[nb] < T) next.push(nb);
        }
      }
    }
    q = next;
  }
  return [standing.map(x => x ? '1' : '0').join(''), days];
}

const input = fs.readFileSync(0,'utf8').trim().split(/\s+/);
if (input.length === 0) {
  const [final, days] = simulateFalling("1101101", 1);
  console.log(final);
  console.log(days);
} else {
  const s = input[0];
  const T = parseInt(input[1], 10);
  const [final, days] = simulateFalling(s, T);
  console.log(final);
  console.log(days);
}
